### PR TITLE
hw-online-book-store

### PR DESCRIPTION
**Pull request by savchenkoarts**

### CODE REVIEW

File: src/main/java/mate/academy/onlinebookstore/repository/BookRepositoryImpl.java [hunk 1]
```diff
@@ -0,0 +1,17 @@
+package mate.academy.onlinebookstore.repository;
+
+import java.util.List;
+import mate.academy.onlinebookstore.model.Book;
+
+public class BookRepositoryImpl implements BookRepository {
+
+    @Override
+    public Book save(Book book) {
+        return null;
```
Comment (Rommelua) [inline src/main/java/mate/academy/onlinebookstore/repository/BookRepositoryImpl.java:10]: implement methods

File: src/main/java/mate/academy/onlinebookstore/service/BookServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+public class BookServiceImpl implements BookService {
+    @Override
+    public Book save(Book book) {
+        return new Book();
+    }
+
+    @Override
+    public List findAll() {
+        return null;
```
Comment (Rommelua) [inline src/main/java/mate/academy/onlinebookstore/service/BookServiceImpl.java:14]: implement methods

File: src/main/java/mate/academy/onlinebookstore/model/Book.java [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+import jakarta.persistence.Entity;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import java.math.BigDecimal;
+import lombok.Data;
+
+@Entity
+@Data
+public class Book {
```
Comment (Rommelua) [inline src/main/java/mate/academy/onlinebookstore/model/Book.java:12]: constraints?
image

**Review by Rommelua — CHANGES_REQUESTED** [review]

File: src/main/java/mate/academy/onlinebookstore/OnlineBookStoreApplication.java [hunk 4]
```diff
@@ -1,13 +1,40 @@
...
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.CommandLineRunner;
 import org.springframework.boot.SpringApplication;
 import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.context.annotation.Bean;
 
 @SpringBootApplication
 public class OnlineBookStoreApplication {
+    @Autowired
+    private BookService bookService;
```
Comment (sokimaaa) [inline src/main/java/mate/academy/onlinebookstore/OnlineBookStoreApplication.java:15]: Wrong way to inject bean.
Inject directly via method params in the @Bean method.

```suggestion
```

File: src/main/java/mate/academy/onlinebookstore/OnlineBookStoreApplication.java [hunk 5]
```diff
@@ -1,13 +1,40 @@
...
+                book.setDescription("About nature");
+                book.setTitle("From nature");
+                book.setPrice(BigDecimal.valueOf(500));
+                book.setAuthor("Stiven");
+
+                bookService.save(book);
+
+                System.out.println(bookService.findAll());
+            }
+        };
```
Comment (sokimaaa) [inline src/main/java/mate/academy/onlinebookstore/OnlineBookStoreApplication.java:38]: use lambda (not anonymous class)
```suggestion
        return new CommandLineRunner() {
            @Override
            public void run(String... args) throws Exception {
                Book book = new Book();
                book.setCoverImage("image");
                book.setIsbn("Ranok");
                book.setDescription("About nature");
                book.setTitle("From nature");
                book.setPrice(BigDecimal.valueOf(500));
                book.setAuthor("Stiven");

                bookService.save(book);

                System.out.println(bookService.findAll());
            }
        };
```

File: src/main/java/mate/academy/onlinebookstore/model/Book.java [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+    @Column(nullable = false)
+    private String title;
+    @NotBlank
+    @Column(nullable = false)
+    private String author;
+    @NotBlank
+    @Column(nullable = false, unique = true)
+    private String isbn;
+    @NotNull
+    @Column(nullable = false)
```
Comment (sokimaaa) [inline src/main/java/mate/academy/onlinebookstore/model/Book.java:29]: notNull, notBlank are not `constraints`. Remove annotations that you don't know how to use. 
```suggestion
    @Column(nullable = false)
    private String title;
    @Column(nullable = false)
    private String author;
    @Column(nullable = false, unique = true)
    private String isbn;
    @Column(nullable = false)
```

**Pull request merged**